src/com/aca/homework/week6/buffer/BufferImpl.java

```diff
@@ -0,0 +1,50 @@
+package com.aca.homework.week6.buffer;
+
+import java.util.Objects;
+
+public class BufferImpl implements Buffer {
+    private Object value;
+
+    public void put(Object value) {
+        if (!isBufferEmpty()) {
+            System.out.println("The buffer is not empty");
+            return;
+        }
+        this.value = value;
+
+    }
+
+    public Object get() {
+        if (isBufferEmpty()) {
+            System.out.println("The buffer is empty");
```

Comment:
you can return null here to not do other unnecessary steps even if the result is the same

Reply:
Done